File: JavaScript_Courses/JavaScript_Basics/Preparation/06_Nested_Loops_Exercise/03_Sum_Prime_Non_Prime.js
function sumPrimeNonPrime(input) {
    let sumPrime = 0;
    let sumNonPrime = 0;

    let i = 0;
    let currentCommand = input[i];

    while (currentCommand != "stop") {
        currentNum = Number(currentCommand);

        if (currentNum < 0) {
            console.log("Number is negative.");
            i++;
            currentCommand = input[i];
            continue;
        }

        let isPrime = true;
        for (let j = 2; j < currentNum; j++) {
            if (currentNum % j == 0) {
                isPrime = false;
                break;
            }
        }

        if (isPrime) {
            sumPrime += currentNum;
        } else {
            sumNonPrime += currentNum;
        }

        i++;
        currentCommand = input[i];
    }

    console.log(`Sum of all prime numbers is: ${sumPrime}`);
    console.log(`Sum of all non prime numbers is: ${sumNonPrime}`);
}


// sumPrimeNonPrime([
//     "3",
//     "9",
//     "0",
//     "7",
//     "19",
//     "4",
//     "stop",
// ]);

// sumPrimeNonPrime([
//     "30",
//     "83",
//     "33",
//     "-1",
//     "20",
//     "stop",
// ]);

File: Python_Courses/Python_Basics/Python_PB_2024_10/06_Nested_Loops_Exercise/03_Sum_Prime_Non_Prime.py
# •	"Sum of all prime numbers is: {prime numbers sum}"
# •	"Sum of all non prime numbers is: {nonprime numbers sum}"

# "stop"
prime_numbers_sum = 0
nonprime_numbers_sum = 0

command = input()
while command != 'stop':  # 'stop' OR number_as_str ('9')
    current_number = int(command)

    if current_number < 0:
        print('Number is negative.')
        command = input()
        continue

    is_prime = True
    for divisor in range(2, current_number):  # [2, 3, 4, ... number-1]  == [2, ]  # [2, ... 8]
        if current_number % divisor == 0:
            is_prime = False
            break

    if is_prime:  # is_prime [True, False]
        prime_numbers_sum += current_number
    else:
        nonprime_numbers_sum += current_number

    command = input()

print(f'Sum of all prime numbers is: {prime_numbers_sum}')
print(f'Sum of all non prime numbers is: {nonprime_numbers_sum}')
